app/lib/rocketchat.js

```diff
@@ -836,16 +836,20 @@ const RocketChat = {
 		return this.post('teams.removeRoom', { roomId, teamId });
 	},
 	leaveTeam({ teamName, rooms }) {
+		const serverVersion = reduxStore.getState().server.version;
 		// RC 3.13.0
-		return this.post('teams.leave', { teamName, rooms });
+		return this.post('teams.leave', {
+			teamName,
+			...((compareServerVersion(serverVersion, '4.2', methods.lowerThan) || rooms?.length) && { rooms })
```

Comment:
I'm sorry, but I didn't get this logic.
- If the server is 4.1, it's going to use `rooms`
- If `rooms` are send, it's going to use `rooms`
So the only way not to use rooms, is server 4.2+ and not selecting any room on leave, but then why testing the server?
Isn't testing `rooms?.length` enough?

Reply:
For servers lower than 4.2.0 we must send rooms as param, at least an empty array. However the server 4.2.0 expects a minimum of 1 item inside the rooms' array, but he didn't accept if the array is empty, so if the array is empty we can't pass rooms as param.